Mathlib/Algebra/Hom/Centroid.lean

```diff
@@ -0,0 +1,532 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Christopher Hoskin
+
+! This file was ported from Lean 3 source module algebra.hom.centroid
+! leanprover-community/mathlib commit 6cb77a8eaff0ddd100e87b1591c6d3ad319514ff
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.GroupPower.Lemmas
+import Mathlib.Algebra.Hom.GroupInstances
+
+/-!
+# Centroid homomorphisms
+
+Let `A` be a (non unital, non associative) algebra. The centroid of `A` is the set of linear maps
+`T` on `A` such that `T` commutes with left and right multiplication, that is to say, for all `a`
+and `b` in `A`,
+$$
+T(ab) = (Ta)b, T(ab) = a(Tb).
+$$
+In mathlib we call elements of the centroid "centroid homomorphisms" (`centroid_hom`) in keeping
+with `add_monoid_hom` etc.
+
+We use the `fun_like` design, so each type of morphisms has a companion typeclass which is meant to
+be satisfied by itself and all stricter types.
+
+## Types of morphisms
+
+* `centroid_hom`: Maps which preserve left and right multiplication.
+
+## Typeclasses
+
+* `centroid_hom_class`
+
+## References
+
+* [Jacobson, Structure of Rings][Jacobson1956]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+## Tags
+
+centroid
+-/
+
+-- Porting note: This is so that `instance : Ring (CentroidHom α)` doesn't time out.
+set_option maxHeartbeats 1000000
```

Comment:
Please move it to that particular instance using
```
set_option ... in
```
syntax

Reply:
I'm not exactly sure what you mean; is the most recent commit what you're looking for?